client/src/main/java/org/apache/celeborn/client/ShuffleClient.java

```diff
@@ -147,11 +147,21 @@ public abstract int mergeData(
   public abstract void pushMergedData(String applicationId, int shuffleId, int mapId, int attemptId)
       throws IOException;
 
-  // Report partition locations written by the completed map task
+  // Report partition locations written by the completed map task of ReducePartition Shuffle Type
   public abstract void mapperEnd(
       String applicationId, int shuffleId, int mapId, int attemptId, int numMappers)
       throws IOException;
 
+  // Report partition locations written by the completed map task of MapPartition Shuffle Type
+  public abstract void mapPartitionMapperEnd(
+      String applicationId,
+      int shuffleId,
+      int mapId,
```

Comment:
Do we need mapId? PartitionId already contains mapId

Reply:
> Do we need mapId? PartitionId already contains mapId

Not so necessary, But to avoid lifecycle manager to understand the semantics of partitionId. I think add mapId would be better.

Reply:
Then